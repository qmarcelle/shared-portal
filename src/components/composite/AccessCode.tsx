import Image from 'next/image';
import AlertIcon from '../../../public/assets/alert_gray.svg';
import { Column } from '../foundation/Column';
import { Row } from '../foundation/Row';
import { TextBox } from '../foundation/TextBox';

export type AccessCodeProps = {
  accessCodeData: string;
};
export const AccessCode = ({ accessCodeData }: AccessCodeProps) => {
  return (
    <Column>
      {accessCodeData && (
        <div className="access-code ">
          <TextBox text="Access Code" className="ml-6 mt-3 text-[10px]" />

          <TextBox
            text={accessCodeData}
            className="ml-3  font-bold text-[15px]"
          />
        </div>
      )}
      {!accessCodeData && (
        <Row className="card-neutral h-[44px]">
          <Image
            src={AlertIcon}
            className="h-[19px] w-[19px] mt-3 ml-3"
            alt="alert"
          />
          <TextBox text="Access code could not load." className="mt-3 ml-2" />
        </Row>
      )}
    </Column>
  );
};
