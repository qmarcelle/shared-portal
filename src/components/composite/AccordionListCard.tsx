import { AccordionInfoItem } from '@/models/accordionInfoItem';
import { IComponent } from '../IComponent';
import { Accordion } from '../foundation/Accordion';
import { Card } from '../foundation/Card';
import { Column } from '../foundation/Column';
import { Divider } from '../foundation/Divider';
import { Header } from '../foundation/Header';
import { Spacer } from '../foundation/Spacer';

interface AccordionListCardProps extends IComponent {
  header: string;
  information: AccordionInfoItem[];
}

export const AccordionListCard = ({
  header,
  information,
  className,
}: AccordionListCardProps) => {
  return (
    <Card className={`${className ?? ''} ${header ? 'large-section' : ''}`}>
      <Column className="items-stretch">
        {header && (
          <div>
            <Header className="title-2" text={header}></Header>
            <Spacer size={22} />
          </div>
        )}
        {information.map((info, index) => (
          <>
            <Accordion
              key={index}
              className="px-2 py-4 "
              label={info.title}
              icon={info.icon}
              child={info.body}
              initialOpen={false}
              type="card"
              openIcon={<Image className="pl-2 w-6" src={Down} alt=""></Image>}
              closeIcon={<Image className="pl-2 w-6" src={Up} alt=""></Image>}
              onOpenCallBack={() =>
                info.onOpenCallBack && info.onOpenCallBack()
              }
            ></Accordion>
            {index !== information.length - 1 && <Divider />}
          </>
        ))}
      </Column>
    </Card>
  );
};
