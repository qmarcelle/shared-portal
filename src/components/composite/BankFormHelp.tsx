import { ChatTrigger } from '@/app/clicktochat/components/ChatTrigger';
import { Card } from '../foundation/Card';
import { Column } from '../foundation/Column';
import { Header } from '../foundation/Header';
import { RichText } from '../foundation/RichText';
import { Spacer } from '../foundation/Spacer';

export const BankFormHelp = () => (
  <Column>
    <Card type="elevated" className="small-section">
      <Column>
        <Header className="title-2" text="Get Help with the Bank Draft Form" />
        <Spacer size={16} />
        <RichText
          spans={[
            <span key="1">
              If you need help, please reach out to us. You can{' '}
            </span>,
            <span key="2" className="link font-bold">
              <ChatTrigger>start a chat</ChatTrigger>
            </span>,
            <span key="3"> or call us at [1-800-000-000].</span>,
          ]}
        />
      </Column>
    </Card>
  </Column>
);
