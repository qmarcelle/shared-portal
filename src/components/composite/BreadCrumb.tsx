import '@/styles/breadcrumb.css';
import Image from 'next/image';
import Link from 'next/link';
import { rightWhiteIcon } from '../foundation/Icons';
import { IComponent } from '../IComponent';
export interface BreadcrumbItem {
  path: string;
  label: string;
}

export interface BreadcrumbProps extends IComponent {
  items: BreadcrumbItem[];
}

export const BreadCrumb = ({ items }: BreadcrumbProps) => {
  const current = items.length - 2;

  function computePathLink(index: number) {
    return items
      .slice(0, index - (items.length-1))
      .map((item) => item.path)
      .join("");
  }

  return (
    <nav aria-label="Breadcrumb" className="flex items-center">
      <ol className="breadcrumb items-center body-2 ">
        <>
          {items.map((item, index) => (
            <li key={index} className="hidden md:flex items-center ml-0">
              {index === items.length - 1 ? (
                <span aria-current="page" className="breadcrumb-current">
                  {item.label}
                </span>
              ) : (
                <>
                  <Link
                    className="text-primary font-bold"
                    href={computePathLink(index)}
                  >
                    {item.label}
                  </Link>
                  <div className="px-2">
                    <Image
                      src={rightWhiteIcon}
                      alt=""
                      className="mx-1 h-4 w-4"
                    />
                  </div>
                </>
              )}
            </li>
          ))}
        </>
      </ol>
    </nav>
  );
};

