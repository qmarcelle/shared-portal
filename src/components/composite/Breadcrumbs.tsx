'use client';

import { Breadcrumb } from '@/models/app/breadcrumb';
import '@/styles/breadcrumb.css';
import Image from 'next/image';
import Link from 'next/link';
import React from 'react';
import { rightWhiteIcon } from '../foundation/Icons';

export interface BreadcrumbsProps {
  breadcrumbs: Breadcrumb[];
}

const Breadcrumbs: React.FC<BreadcrumbsProps> = ({
  breadcrumbs,
}: BreadcrumbsProps) => {
  return (
    <nav aria-label="Breadcrumb" className="flex items-center">
      <ol className="breadcrumb items-center body-2 ">
        <>
          {breadcrumbs.map((breadcrumb, index) => (
            <li key={index} className="flex items-center ml-0">
              {index === breadcrumbs.length - 1 ? (
                <span aria-current="page" className="breadcrumb-current">
                  {breadcrumb.title}
                </span>
              ) : (
                <>
                  <Link
                    className="text-primary font-bold"
                    href={breadcrumb.path}
                  >
                    {breadcrumb.title}
                  </Link>
                  <div className="px-2">
                    <Image
                      src={rightWhiteIcon}
                      alt=""
                      className="mx-1 h-4 w-4"
                    />
                  </div>
                </>
              )}
            </li>
          ))}
        </>
      </ol>
    </nav>
  );
};

export default Breadcrumbs;
