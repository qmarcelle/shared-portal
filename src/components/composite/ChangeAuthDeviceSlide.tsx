import { googleAnalytics } from '@/utils/analytics';
import { ReactElement } from 'react';
import { Button } from '../foundation/Button';
import { Column } from '../foundation/Column';
import { Header } from '../foundation/Header';
import { Spacer } from '../foundation/Spacer';
import { TextBox } from '../foundation/TextBox';

/* eslint-disable @typescript-eslint/no-explicit-any */
interface ChangeAuthDeviceProps {
  label: string;
  subLabel: string;
  actionArea: ReactElement;
  bottomNote?: string;
  nextCallback: (() => any) | undefined;
  cancelCallback: () => any;
}

export const ChangeAuthDeviceSlide = ({
  label,
  subLabel,
  actionArea,
  bottomNote,
  nextCallback,
  cancelCallback,
}: ChangeAuthDeviceProps) => {
  return (
    <Column className="items-center">
      <Header className="title-2" text={label} />
      <Spacer size={16} />
      <TextBox className="text-center" text={subLabel} />
      <Spacer size={32} />
      <Column className="w-[352px]">
        {actionArea}
        <Spacer size={24} />
        {bottomNote && <TextBox className="mb-8" text={bottomNote} />}
        <Button
          callback={() => {
            nextCallback?.();
            googleAnalytics(
              'next',
              undefined,
              label,
              'next',
              'select_content',
              'select',
            );
          }}
          label="Next"
        />
        <Spacer size={16} />
        <Button
          callback={() => {
            cancelCallback?.();
            googleAnalytics(
              'cancel',
              undefined,
              label,
              'cancel',
              'select_content',
              'select',
            );
          }}
          type="ghost"
          label="Cancel"
        />
      </Column>
    </Column>
  );
};
