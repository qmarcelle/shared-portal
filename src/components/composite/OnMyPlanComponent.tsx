import { Column } from '@/components/foundation/Column';
import { Spacer } from '@/components/foundation/Spacer';
import { OnMyPlanDetails } from '@/models/onmyplan_details';
import { ReactElement } from 'react';
import { IComponent } from '../IComponent';
import { AppLink } from '../foundation/AppLink';
import { OnMyPlanItem } from './OnMyPlanItem';

interface OnMyPlanDropDownProps extends IComponent {
  onMyPlanDetails: OnMyPlanDetails[];
  header?: ReactElement;
  subHeader?: ReactElement;
  planType?: ReactElement;
  infoIcon: boolean;
  allowUpdates?: boolean;
}

export const OnMyPlanComponent = ({
  onMyPlanDetails,
  header,
  subHeader,
  planType,
  infoIcon,
  allowUpdates = true,
}: OnMyPlanDropDownProps) => {
  return (
    <Column className="flex flex-col">
      {header}
      {subHeader && (
        <div>
          {' '}
          <Spacer size={16} />
          {subHeader}
          <Spacer size={32} />
        </div>
      )}
      {planType && <div>{planType}</div>}
      <Spacer size={32} />
      <Column className="flex flex-col">
        {onMyPlanDetails.map((item, index) => (
          <OnMyPlanItem
            key={index}
            className="mb-4"
            memberName={item.memberName}
            DOB={item.DOB}
            sharingType={item.sharingType}
            isMinor={item.isMinor}
            infoButton={infoIcon}
            requestorType={item.requestorType}
            targetType={item.targetType}
            medicalEffectiveDate={item.medicalEffectiveDate}
            dentalEffectiveDate={item.dentalEffectiveDate}
            visionEffectiveDate={item.visionEffectiveDate}
            allowUpdates={allowUpdates}
          />
        ))}
      </Column>
      <Spacer size={32} />
      <AppLink label="View Benefits & Coverage" />
    </Column>
  );
};
