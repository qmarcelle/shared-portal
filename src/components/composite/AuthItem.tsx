import Image from 'next/image';
import AlertIcon from '/assets/alert_gray.svg';
import DentalIcon from '/assets/dental.svg';
import MedicalIcon from '/assets/medical.svg';
import PharmacyIcon from '/assets/pharmacy.svg';
import VisionIcon from '/assets/vision.svg';
import { PriorAuthorizationHelp } from '../composite/PriorAuthorizationHelp';
import { Card } from '../foundation/Card';
import { Column } from '../foundation/Column';
import { Divider } from '../foundation/Divider';
import { Header } from '../foundation/Header';
import { Row } from '../foundation/Row';
import { Spacer } from '../foundation/Spacer';
import { StatusLabel } from '../foundation/StatusLabel';
import { TextBox } from '../foundation/TextBox';
import { IComponent } from '../IComponent';

interface priorAuthDetailProps extends IComponent {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  authInfo: any;
}

export const AuthItem = ({ authInfo }: priorAuthDetailProps) => {
  function getSuccessStatus() {
    switch (authInfo.priorAuthDetailStatus) {
      case 'Processed':
        return 'success';
      case 'Denied':
        return 'error';
      case 'Pending':
        return 'neutral';
      case 'PartialApproval':
        return 'partialapproval';
      case 'Approved':
        return 'success';
      default:
        return 'empty';
    }
  }

  function getClaimIcon() {
    if (authInfo.priorAuthDetailType == 'Medical') {
      return MedicalIcon;
    } else if (authInfo.priorAuthDetailType == 'Dental') {
      return DentalIcon;
    } else if (authInfo.priorAuthDetailType == 'Vision') {
      return VisionIcon;
    } else {
      return PharmacyIcon;
    }
  }

  return (
    <main className="flex flex-col justify-center items-center page priorAuthDetail">
      <Spacer size={32}></Spacer>
      <Column className={'app-content app-base-font-color m-5'}>
        <Row className="md:max-my-8">
          <Image
            src={getClaimIcon()}
            className="icon !w-10 !h-10 w-fit"
            alt={authInfo.priorAuthDetailType}
          />
          <Spacer axis="horizontal" size={8} />
          <Header
            type="title-3"
            text={authInfo.priorAuthDetailName}
            className="md:text-3xl !font-light ml-5 w-3/5 mb-4"
          />
        </Row>
        <Column className="statusDetail">
          <StatusLabel
            label={authInfo.priorAuthDetailStatus}
            status={getSuccessStatus()}
          />
          <TextBox
            type="body-1"
            className="mt-4"
            text={'Visited on ' + authInfo.dateOfVisit}
          ></TextBox>
          <TextBox
            type="body-1"
            className="mt-4"
            text={'For ' + authInfo.member}
          ></TextBox>
          <TextBox
            type="body-1"
            className="mt-4"
            text={'Reference ID : ' + authInfo.PriorAuthReferenceId}
          ></TextBox>
        </Column>
      </Column>
      <Column className="app-content app-base-font-color">
        <section className="flex flex-row items-start app-body">
          <Column className="flex-grow page-section-63_33 items-stretch">
            {(authInfo.priorAuthDetailStatus === 'PartialApproval' ||
              authInfo.priorAuthDetailStatus === 'Denied') && (
              <Card className="neutral container mb-4">
                <Column className="flex flex-row align-top m-4">
                  <Row>
                    <Image src={AlertIcon} className="icon" alt="alert" />
                    <Spacer axis="horizontal" size={8} />
                    <TextBox
                      type="body-1"
                      className="font-bold"
                      text="Get Help Understanding Partial Approvals."
                    ></TextBox>
                  </Row>

                  <Row className="flex flex-col flex-grow mt-4">
                    <TextBox
                      type="body-1"
                      text="We've been able to approve part of this request. We will
                      send you a letter explaining why and details on how to ask
                      for an appeal."
                    ></TextBox>
                    <TextBox
                      type="body-1"
                      className="mt-4"
                      text="For more information,Please start to chart or call
                      [1-800-000-000]."
                    ></TextBox>
                  </Row>
                </Column>
              </Card>
            )}
            <Card className="mb-8 p-8">
              <Column>
                <Header type="title-2" text="Service Details" />
                <Spacer size={16} />
                <Column className="mb-8 md:flex-row">
                  <Column className="mr-7">
                    <Header type="title-3" text="Referred By:" />
                    {(authInfo.priorAuthDetailStatus === 'PartialApproval' ||
                      authInfo.priorAuthDetailStatus === 'Approved') && (
                      <Card className="neutral container mt-4">
                        <Row className="flex flex-row align-top m-4">
                          <TextBox
                            className="w-4/5"
                            type="body-1"
                            text="Provider Information Unavailable."
                          ></TextBox>
                        </Row>
                      </Card>
                    )}
                    {(authInfo.priorAuthDetailStatus === 'Denied' ||
                      authInfo.priorAuthDetailStatus === 'Pending') && (
                      <Column className="w-52">
                        <TextBox
                          type="body-1"
                          className="font-bold mt-4"
                          text={authInfo.referredName}
                        ></TextBox>
                        <TextBox
                          type="body-2"
                          className="mt-4"
                          text="Facility Address"
                        ></TextBox>
                        <TextBox
                          type="body-1"
                          className="mt-4"
                          text="123 Street Address Road City Town, TN 12345"
                        ></TextBox>

                        <TextBox
                          type="body-2"
                          className="mt-4"
                          text="Phone"
                        ></TextBox>
                        <TextBox
                          type="body-1"
                          className="mt-4"
                          text="(123) 456-7890"
                        ></TextBox>
                      </Column>
                    )}
                  </Column>
                  <Column>
                    <Header type="title-3" text="Referred To:" />
                    {(authInfo.priorAuthDetailStatus === 'Denied' ||
                      authInfo.priorAuthDetailStatus === 'Approved') && (
                      <Card className="neutral container mt-4">
                        <Row className="flex flex-row align-top m-4">
                          <TextBox
                            type="body-1"
                            className="w-4/5"
                            text="Provider Information Unavailable."
                          ></TextBox>
                        </Row>
                      </Card>
                    )}
                    {(authInfo.priorAuthDetailStatus === 'PartialApproval' ||
                      authInfo.priorAuthDetailStatus === 'Pending') && (
                      <Column className="w-52">
                        <TextBox
                          type="body-1"
                          className="font-bold mt-4"
                          text={authInfo.referredName}
                        ></TextBox>
                        <TextBox
                          type="body-2"
                          className="mt-4"
                          text="Facility Address"
                        ></TextBox>
                        <TextBox
                          type="body-1"
                          className="mt-4"
                          text="123 Street Address Road City Town, TN 12345"
                        ></TextBox>

                        <TextBox
                          type="body-2"
                          className="mt-4"
                          text="Phone"
                        ></TextBox>
                        <TextBox
                          type="body-1"
                          className="mt-4"
                          text="(123) 456-7890"
                        ></TextBox>
                      </Column>
                    )}
                  </Column>
                </Column>
                <Divider />
                <Row>
                  {(authInfo.priorAuthDetailStatus === 'PartialApproval' ||
                    authInfo.priorAuthDetailStatus === 'Approved' ||
                    authInfo.priorAuthDetailStatus === 'Denied') && (
                    <Column className="w-52">
                      <Header
                        type="title-3"
                        text="Provider Facility"
                        className="my-8"
                      />
                      <TextBox
                        type="body-1"
                        className="font-bold"
                        text="Local Care Hospital"
                      ></TextBox>
                      <TextBox
                        type="body-2"
                        className="mt-4"
                        text="Facility Address"
                      ></TextBox>
                      <TextBox
                        type="body-1"
                        className="mt-4"
                        text="123 Street Address Road City Town, TN 12345"
                      ></TextBox>

                      <TextBox
                        type="body-2"
                        className="mt-4"
                        text="Phone"
                      ></TextBox>
                      <TextBox
                        type="body-1"
                        className="mt-4"
                        text="(123) 456-7890"
                      ></TextBox>
                    </Column>
                  )}
                  <Column>
                    {authInfo.priorAuthDetailStatus === 'Pending' && (
                      <Card className="neutral container mt-4">
                        <Row className="flex flex-row align-top m-4">
                          <TextBox
                            className="w-4/5"
                            type="body-1"
                            text="Provider Information Unavailable."
                          ></TextBox>
                        </Row>
                      </Card>
                    )}
                  </Column>
                </Row>
              </Column>
            </Card>
          </Column>
          <Column className=" flex-grow page-section-36_67 items-stretch md:mt-0">
            <PriorAuthorizationHelp />
          </Column>
        </section>
        <Spacer size={32}></Spacer>
      </Column>
    </main>
  );
};
