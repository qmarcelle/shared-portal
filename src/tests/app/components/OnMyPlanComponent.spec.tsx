import { OnMyPlanComponent } from '@/components/composite/OnMyPlanComponent';
import { render, screen } from '@testing-library/react';

const renderUI = () => {
  render(
    <OnMyPlanComponent
      infoIcon={true}
      onMyPlanDetails={[
        {
          memberName: 'Chris Hall',
          DOB: '01/01/1978',
          sharingType: 'Medical / Dental / Vision',
          isMinor: false,
        },
        {
          memberName: 'Forest Hall',
          DOB: '01/01/2001',
          sharingType: 'Medical',
          isMinor: false,
        },
        {
          memberName: 'Corey Hall',
          DOB: '01/01/2002',
          sharingType: 'Medical',
          isMinor: false,
        },
        {
          memberName: 'Telly Hall',
          DOB: '01/01/2008',
          sharingType: 'Medical',
          isMinor: false,
        },
      ]}
    />,
  );
};

describe('OnMyPlanComponent', () => {
  it('should render the UI correctly', async () => {
    const component = renderUI();
    screen.getByText('Chris Hall');
    screen.getByText('DOB: 01/01/1978');
    screen.getByText('Medical / Dental / Vision');

    expect(component).toMatchSnapshot();
  });
});
