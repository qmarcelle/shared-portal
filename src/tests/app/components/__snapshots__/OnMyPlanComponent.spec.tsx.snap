// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnMyPlanComponent should render the UI correctly 1`] = `undefined`;
